staging/src/k8s.io/apimachinery/pkg/util/strategicpatch/testing/fake_openapi.go

```diff
@@ -0,0 +1,160 @@
+}
+
+type fakeOpenapiResources struct {
+	openapi.Definitions
+	mergeItem          openapi.Schema
+	retainKeyMergeItem openapi.Schema
+	precisionItem      openapi.Schema
+}
+
+func NewOpenAPIData(doc *openapi_v2.Document) (*fakeOpenapiResources, error) {
+	definitions := fakeOpenapiResources{
+		Definitions: openapi.Definitions{
+			Models:    map[string]openapi.Schema{},
+			Resources: map[schema.GroupVersionKind]string{},
+		},
+	}
+
+	// Save the list of all models first. This will allow us to
+	// validate that we don't have any dangling reference.
+	for _, namedSchema := range doc.GetDefinitions().GetAdditionalProperties() {
+		definitions.Models[namedSchema.GetName()] = nil
+	}
+
+	// Now, parse each model. We can validate that references exists.
+	for _, namedSchema := range doc.GetDefinitions().GetAdditionalProperties() {
+		path := openapi.NewPath(namedSchema.GetName())
+		schema, err := definitions.ParseSchema(namedSchema.GetValue(), &path)
+		if err != nil {
+			return nil, err
+		}
+		definitions.Models[namedSchema.GetName()] = schema
+		if strings.Contains(namedSchema.GetName(), "mergeItem") {
+			definitions.mergeItem = schema
+		}
+		if strings.Contains(namedSchema.GetName(), "retainKeyMergeItem") {
+			definitions.retainKeyMergeItem = schema
+		}
+		if strings.Contains(namedSchema.GetName(), "precisionItem") {
+			definitions.precisionItem = schema
+		}
+	}
+
+	return &definitions, nil
+}
+
+func (r *fakeOpenapiResources) getMergeItem() (openapi.Schema, error) {
+	if r.mergeItem == nil {
+		return nil, errors.New("cannot find mergeItem in openapi schema.")
+	}
+	if k, ok := r.mergeItem.(*openapi.Kind); !ok {
+		return nil, mergepatch.ErrBadArgType(k, r.mergeItem)
+	}
+	return r.mergeItem, nil
+}
+
+func (r *fakeOpenapiResources) getRetainKeysMergeItem() (openapi.Schema, error) {
+	if r.retainKeyMergeItem == nil {
+		return nil, errors.New("cannot find retainKeyMergeItem in openapi schema.")
+	}
+	if k, ok := r.retainKeyMergeItem.(*openapi.Kind); !ok {
+		return nil, mergepatch.ErrBadArgType(k, r.retainKeyMergeItem)
+	}
+	return r.retainKeyMergeItem, nil
+}
+
+func (r *fakeOpenapiResources) getPrecisionItem() (openapi.Schema, error) {
+	if r.precisionItem == nil {
+		return nil, errors.New("cannot find precisionItem in openapi schema.")
+	}
+	if k, ok := r.precisionItem.(*openapi.Kind); !ok {
+		return nil, mergepatch.ErrBadArgType(k, r.precisionItem)
+	}
+	return r.precisionItem, nil
+}
+
+func GetMergeItemOrDie(openapiDir string) openapi.Schema {
+	oa, err := readOpenapiFromFile(openapiDir)
+	if err != nil {
+		panic(err)
+	}
+	mi, err := oa.getMergeItem()
+	if err != nil {
+		panic(err)
+	}
+	return mi
+}
+
+func GetRetainKeysMergeItemOrDie(openapiDir string) openapi.Schema {
+	oa, err := readOpenapiFromFile(openapiDir)
+	if err != nil {
+		panic(err)
+	}
+	rkmi, err := oa.getRetainKeysMergeItem()
+	if err != nil {
+		panic(err)
+	}
+	return rkmi
+}
+
+func GetPrecisionItemOrDie(openapiDir string) openapi.Schema {
```

Comment:
Not sure how this is used, but it seems strange that we read the entire file to get one single field.